risc0/zkvm/platform/src/syscall.rs

```diff
@@ -245,12 +282,116 @@ pub unsafe fn sys_sha_buffer(
     unimplemented!()
 }
 
-#[inline(always)]
-pub unsafe fn sys_rand(recv_buf: *mut u32, words: usize) {
-    #[cfg(target_os = "zkvm")]
-    {
-        sys_io(recv_buf, words, ptr::null_mut(), 0, SENDRECV_CHANNEL_RANDOM);
+pub unsafe extern "C" fn sys_rand(recv_buf: *mut u32, words: usize) {
+    syscall_0(nr::SYS_RANDOM, recv_buf, words);
+}
+
+pub unsafe extern "C" fn sys_panic(msg_ptr: *const u8, len: usize) -> ! {
+    syscall_2(nr::SYS_PANIC, null_mut(), 0, msg_ptr as u32, len as u32);
+    unreachable!()
+}
+
+pub unsafe extern "C" fn sys_log(msg_ptr: *const u8, len: usize) {
+    syscall_2(nr::SYS_LOG, null_mut(), 0, msg_ptr as u32, len as u32);
+}
+
+pub extern "C" fn sys_cycle_count() -> usize {
+    unsafe {
+        let Return(a0, _) = syscall_0(nr::SYS_CYCLE_COUNT, null_mut(), 0);
+        a0 as usize
     }
-    #[cfg(not(target_os = "zkvm"))]
-    unimplemented!()
+}
+
+/// Returns the number of bytes currently available for reading on the
+/// given file descriptor.
+pub unsafe extern "C" fn sys_read_avail(fd: u32) -> usize {
```

Comment:
What is the expected behavior this call for streaming file descriptors where a length may not be easily available? And for file descriptors with infinite or indeterminate length (e.g. reading from `/dev/urandom`)?

Reply:
Are the semantics here "available now" or "available until end of file"? E.g. Is it acceptable for a call to `SYS_READ_AVAIL` to return `0` and then when called a second time return `1`?

Reply:
"currently available", so available now.  I also unexported sys_read_avail as I didn't mean to expose it to users.

sys_read has standard posix read semantics, so updated the comment.